### ISSUE TITLE
Math.round() gives incorrect results

### ISSUE THREAD

**Issue by jsobell** [nilproject/NiL.JS#117]
After spending a week rewriting the Date object because moment timezone gave incorrect results, it turns out it was caused by an error in the Math.round() function!
A fixed version is here: https://github.com/questmetrics/NiL.JS/blob/develop/NiL.JS/BaseLibrary/Math.cs#L552
The code was assigning the result to an integer, which has insufficient bits to hold large double numbers, so the result was truncated.

**Comment by nilproject** [comment]
Thanks for this bug! I fixed it yesterday, when saw you fix. My fix: c9ea805cd6db4b9c53ed75f05ff366eef32b76ff
But you fix not fully correct. With you implementation this test fails:
```
Test262 Error: #1: 'var x=+Infinity; Math.round(x) !== +Infinity' at 
File: "tests/sputnik/ch15/15.8/15.8.2/15.8.2.15/S15.8.2.15_A4.js"
```

**Comment by nilproject** [comment]
Also I fixed similar bug in Math.floor(..) and fix mistake in utility function name ("shift to right" was called as shl).
